src/main/java/com/google/cloud/spanner/myadapter/session/SessionState.java

```diff
@@ -29,4 +82,100 @@ public ProtocolStatus getProtocolStatus() {
   public void setProtocolStatus(ProtocolStatus protocolStatus) {
     this.protocolStatus = protocolStatus;
   }
+
+  Map<String, SystemVariable> getVariableMapForScope(SessionVariableScope scope) {
+    switch (scope) {
+      case GLOBAL:
+        return GLOBAL_SETTINGS;
+      case SESSION:
+        return settings;
+      default:
+        throw unknownParamError("scope");
+    }
+  }
+  /**
+   * Sets the value of the specified setting. The new value will be persisted if the current
+   * transaction is committed. The value will be lost if the transaction is rolled back.
+   */
+  public void set(String name, String value, SessionVariableScope scope) {
```

Comment:
Convert names to either lower or upper case to ensure both the cases are covered.

Reply:
Already noticed it and have fixed it.